builtin/mainmenu/content/dlg_contentdb.lua

```diff
@@ -176,72 +170,151 @@ local function load()
+		cell_h = cell_w / 4
+
+		local required_height = math.ceil(num_per_page / row_cells) * (cell_h + cell_spacing) - cell_spacing
+		-- Add 0.1 to be more lenient
+		if required_height <= size.y + 0.1 then
+			break
+		end
+
+		row_cells = row_cells + 1
+	end
+
+	return cell_spacing, row_cells, cell_w, cell_h
+end
+
+
+local function calculate_num_per_page()
+	local window = core.get_window_info()
+	local size = { x = window.max_formspec_size.x, y = window.max_formspec_size.y }
+
+	size.x = size.x - 2
+	size.y = size.y - 3
+
+	local coordToPx = window.size.x / window.max_formspec_size.x / window.real_gui_scaling
+
+	local num_per_page = 12
+	while num_per_page > 2 do
+		local _, _, cell_w, _ = fit_cells(num_per_page, size)
+		if cell_w * coordToPx > 350 then
+			break
+		end
+
+		num_per_page = num_per_page - 1
+	end
+	return num_per_page
+end
+
+
 local function get_formspec(dlgdata)
+	local window_padding = { x = PLATFORM == "Android" and 1 or 0.5, y = PLATFORM == "Android" and 1 or 0.5 }
+	local window = core.get_window_info()
+	local size = { x = window.max_formspec_size.x, y = window.max_formspec_size.y }
+
 	if contentdb.loading then
-		return get_info_formspec(fgettext("Loading..."))
+		return get_info_formspec(size, window_padding, fgettext("Loading..."))
 	end
 	if contentdb.load_error then
-		return get_info_formspec(fgettext("No packages could be retrieved"))
+		return get_info_formspec(size, window_padding, fgettext("No packages could be retrieved"))
 	end
 	assert(contentdb.load_ok)
 
 	contentdb.update_paths()
 
+	local num_per_page = dlgdata.num_per_page
 	dlgdata.pagemax = math.max(math.ceil(#contentdb.packages / num_per_page), 1)
 	if cur_page > dlgdata.pagemax then
 		cur_page = 1
 	end
 
-	local W = 15.75
-	local H = 9.5
-	local formspec = {
-		"formspec_version[6]",
-		"size[15.75,9.5]",
-		core.settings:get_bool("enable_touch") and "padding[0.01,0.01]" or "position[0.5,0.55]",
+	local W = size.x - window_padding.x * 2
+	local H = size.y - window_padding.y * 2
 
-		"style[status,downloading,queued;border=false]",
+	local category_x = 0
+	local number_category_buttons = 4
+	local max_button_w = (W - 0.375 - 0.25 - 7) / number_category_buttons
+	local category_button_w = math.min(max_button_w, 3)
+	local function make_category_button(name, label, selected)
+		category_x = category_x + 1
+		local color = selected and mt_color_green or ""
+		return ("style[%s;bgcolor=%s]button[%f,0;%f,0.8;%s;%s]"):format(name, color,
+				(category_x - 1) * category_button_w, category_button_w, name, label)
+	end
 
-		"container[0.375,0.375]",
-		"field[0,0;7.225,0.8;search_string;;", core.formspec_escape(search_string), "]",
+
+	local selected_type = filter_type
+
+	local search_box_width = W - 0.375 - 0.25 - 2*0.8
+			- number_category_buttons * category_button_w
+	local formspec = {
+		"formspec_version[7]",
+		"size[", size.x, ",", size.y, "]",
+		"padding[0,0]",
+		"bgcolor[#3333;true]",
```

Comment:
setting `#3333` as the non-fullscreen color doesn't have any effect here since fullscreen is true (three more occurrences)

Reply:
Removing this statement makes the fullscreen background transparent: 

![image](https://github.com/user-attachments/assets/8972fbcc-1fc8-4ac0-baa0-320fc6b9cbca)